hazelcast-hibernate53/src/test/java/com/hazelcast/hibernate/HibernateTopicTestSupport.java

```diff
@@ -0,0 +1,60 @@
+package com.hazelcast.hibernate;
+
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.hibernate.entity.AnnotatedEntity;
+import com.hazelcast.hibernate.instance.HazelcastMockInstanceLoader;
+import com.hazelcast.hibernate.local.LocalRegionCache;
+import org.hibernate.SessionFactory;
+import org.hibernate.cache.spi.RegionFactory;
+import org.junit.After;
+import org.junit.Before;
+import org.mockito.Mock;
+
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class HibernateTopicTestSupport extends HibernateTestSupport {
```

Comment:
It started out that way, way back when. It _needs_ to be multiple modules in some form or fashion because you need to compile against several different versions of Hibernate--the caching SPIs change constantly. Initially, that was done by basically copying the entire source to a new module [like when I added support for Hibernate 5.0 and 5.1](https://github.com/hazelcast/hazelcast-hibernate/pull/2) (and it was done that way for earlier Hibernate versions back when `hazelcast-hibernate` was part of the core `hazelcast` repository), but that has a lot of drawbacks (especially around bugfixes, since they need to be applied, sometimes with small variations and sometimes without, to all the different modules).

When the Hibernate 5 code was moved here, the duplication was (partially) resolved by switching to the copying approach you see.

(Note: I don't have any agenda in terms of how the codebase is maintained. I'm just trying to offer a little history since I've worked on Hazelcast's Hibernate code pretty extensively. (I also implemented [4.2 and 4.3 support](https://github.com/hazelcast/hazelcast/pull/2621), in addition to 5.0 and 5.1)